src/core/tsi/transport_security.cc

```diff
@@ -251,6 +251,16 @@ tsi_result tsi_handshaker_result_extract_peer(const tsi_handshaker_result* self,
   return self->vtable->extract_peer(self, peer);
 }
 
+tsi_frame_protector_type tsi_handshaker_result_frame_protector_type(
```

Comment:
How about adding some sanity check to fail-early mismatched cases e.g., user only implemented zero-copy frame protector but mistakenly let `frame_protector_type` API return `TSI_FRAME_PROTECTOR_NORMAL`?  which makes me wonder if we should let the API return `tsi_result` and have an output parameter with type of `tsi_frame_protector_type*`?